packages/mui-material/src/InputBase/InputBase.js

```diff
@@ -89,7 +89,7 @@ const useUtilityClasses = (ownerState) => {
       disabled && 'disabled',
       type === 'search' && 'inputTypeSearch',
       multiline && 'inputMultiline',
-      size === 'small' && 'inputSizeSmall',
+      size !== 'medium' && `inputSize${capitalize(size)}`,
```

Comment:
I think we need to consider medium value in size here as well and in all other places.